utils/githooks/pre-commit.d/20-flake.sh

```diff
@@ -0,0 +1,50 @@
+#!/bin/sh
+
+# Runs flake8 for the DAOS project.
+# Will first check uncomiited code, then either the entire tree if on master
+# or a diff against master if not.
+#
+# Picks up config files from .flake8
+
+set -u
+
+if ! command -v flake8 > /dev/null 2>&1
+then
+    echo "No flake checking, install flake8 command to improve pre-commit checks"
+    exit 0
+fi
+
+echo Checking uncommitted code with flake.
+git diff -u | flake8 --diff
+
+RC=$?
+if [ $RC -ne 0 ]
```

Comment:
(lint) Did you forget the 'then' for this 'if'? [SC1049]
(lint) Couldn't parse this if expression. Fix to allow more checks. [SC1073]